evals/scabench/prepare.py

```
@@ -0,0 +1,263 @@
+    if dest.exists():
+        shutil.rmtree(dest)
+    dest.mkdir(parents=True)
+    with tarfile.open(fileobj=io.BytesIO(blob), mode="r:gz") as tar:
+        for member in tar.getmembers():
+            # Strip the GitHub-added top-level directory, and refuse traversal.
+            parts = Path(member.name).parts[1:]
+            if not parts:
+                continue
+            target = dest.joinpath(*parts).resolve()
+            if not str(target).startswith(str(dest.resolve())):
+                raise SystemExit(f"Refusing path traversal in archive: {member.name}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)